include/deal.II/lac/solver_gmres.h

```diff
@@ -891,10 +891,10 @@ namespace internal
 
       for (; j < vv.locally_owned_size(); ++j)
         {
-          double temp = vv(j);
+          double temp = vv.local_element(j);
           for (unsigned int i = 0; i < dim; ++i)
-            temp -= h(i) * orthogonal_vectors[i](j);
-          vv(j) = temp;
+            temp -= h(i) * orthogonal_vectors[i].local_element(j);
+          vv.local_element(j) = temp;
 
           norm_vv_temp += temp * temp;
         }
```

Comment:
This looks right, but it confounds me why we have never triggered an exception with this or seen this have any other ill effect. This is, after all, a fairly widely used class. Do you know how this has escaped our attention so far?

Reply:
This is very recent code from #14349, and we in fact do not use it as a default option, so we have simply been lucky in the last month.

Reply:
I see. And yes, a test would be nice!